cmd/fs-v1-metadata.go

```diff
@@ -274,167 +271,119 @@ func newFSMetaV1() (fsMeta fsMetaV1) {
+		return fmt.Errorf("Unable to recognize backend format, Disk is not in FS format. %s", format.Format)
 	}
 
-	return nil
-}
+	// Check if format is currently supported.
+	if format.FS.Version != fsFormatVersion {
+		return errCorruptedFormat
+	}
 
-// This is just kept as reference, there is no sanity
-// check for FS format in version "1".
-func checkFormatSanityFSV1(fsPath string) error {
 	return nil
 }
 
-// Check for sanity of FS format in version "2".
-func checkFormatSanityFSV2(fsPath string) error {
-	buckets, err := readDir(pathJoin(fsPath, minioMetaBucket, bucketConfigPrefix))
-	if err != nil && err != errFileNotFound {
+func loadFormatFS(formatCfg *formatConfigV1, lk *lock.LockedFile) error {
+	_, err := formatCfg.ReadFrom(lk)
+	if err != nil && errorCause(err) != io.EOF {
 		return err
 	}
 
-	// Attempt to validate all the buckets have a sanitized backend.
-	for _, bucket := range buckets {
-		entries, rerr := readDir(pathJoin(fsPath, minioMetaBucket, bucketConfigPrefix, bucket))
-		if rerr != nil {
-			return rerr
-		}
-
-		var expectedConfigs = append(bucketMetadataConfigs, objectMetaPrefix+"/")
-		entriesSet := set.CreateStringSet(entries...)
-		expectedConfigsSet := set.CreateStringSet(expectedConfigs...)
-
-		// Entries found shouldn't be more than total
-		// expected config directories, files.
-		if len(entriesSet) > len(expectedConfigsSet) {
-			return errCorruptedFormat
-		}
-
-		// Look for the difference between entries and the
-		// expected config set, resulting entries if they
-		// intersect with original entries set we know
-		// that the backend has unexpected files.
-		if !entriesSet.Difference(expectedConfigsSet).IsEmpty() {
-			return errCorruptedFormat
-		}
+	// No data on disk `format.json` still empty treat
+	// it as unformatted disk.
+	if errorCause(err) == io.EOF {
+		return errUnformattedDisk
 	}
-	return nil
-}
 
-// Check for sanity of FS format for a given version.
-func checkFormatSanityFS(fsPath string, fsFormatVersion string) (err error) {
-	switch fsFormatVersion {
-	case fsFormatV2:
-		err = checkFormatSanityFSV2(fsPath)
-	default:
-		err = errCorruptedFormat
-	}
-	return err
+	return nil
 }
 
-// Initializes a new `format.json` if not present, validates `format.json`
-// if already present and migrates to newer version if necessary. Returns
-// the final format version.
-func initFormatFS(fsPath, fsUUID string) (err error) {
-	fsFormatPath := pathJoin(fsPath, minioMetaBucket, fsFormatJSONFile)
-
-	// fsFormatJSONFile - format.json file stored in minioMetaBucket(.minio.sys) directory.
-	lk, err := lock.LockedOpenFile(preparePath(fsFormatPath), os.O_RDWR|os.O_CREATE, 0600)
+// Check if disk has already a valid format.
+func checkValidFormatFS(fsFormatPath string) error {
+	rlk, err := lock.RLockedOpenFile(preparePath(fsFormatPath))
 	if err != nil {
+		if os.IsNotExist(err) {
+			// If format.json not found then
+			// its an unformatted disk.
+			return traceError(errUnformattedDisk)
```

Comment:
Its a general practice.. that all low level errors have a traceError() associated for example if at all the caller needs to print the stack they can.  We don't need to return errors with or without stack in the same function just to be consistent.